test/CadlRanchProjectsNonAzure.Tests/type-union.cs

```diff
@@ -0,0 +1,198 @@
+        });
+
+        [Test]
+        public Task SendStringAndArray() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetStringAndArrayClient().SendAsync(new StringAndArrayCases(BinaryData.FromObjectAsJson("test"),
+                BinaryData.FromObjectAsJson(new List<string>() { "test1", "test2" })));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task GetMixedLiterals() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetMixedLiteralsClient().GetMixedLiteralAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            AssertEqual(BinaryData.FromObjectAsJson("a"), response.Value.Prop.StringLiteral);
+            AssertEqual(BinaryData.FromObjectAsJson(2), response.Value.Prop.IntLiteral);
+            AssertEqual(BinaryData.FromObjectAsJson(3.3), response.Value.Prop.FloatLiteral);
+            AssertEqual(BinaryData.FromObjectAsJson(true), response.Value.Prop.BooleanLiteral);
+        });
+
+        [Test]
+        public Task SendMixedLiteralsOnlyOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetMixedLiteralsClient().SendAsync(new MixedLiteralsCases(BinaryData.FromObjectAsJson("a"),
+                BinaryData.FromObjectAsJson(2),
+                BinaryData.FromObjectAsJson(3.3),
+                BinaryData.FromObjectAsJson(true)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task GetMixedTypes() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetMixedTypesClient().GetMixedTypeAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            AssertEqual(BinaryData.FromObjectAsJson(new { name = "test" }), response.Value.Prop.Model);
+            AssertEqual(BinaryData.FromObjectAsJson("a"), response.Value.Prop.Literal);
+            AssertEqual(BinaryData.FromObjectAsJson(2), response.Value.Prop.Int);
+            AssertEqual(BinaryData.FromObjectAsJson(true), response.Value.Prop.Boolean);
+        });
+
+        [Test]
+        public Task SendMixedTypesOnlyOnly() => Test(async (host) =>
+        {
+            var response = await new UnionClient(host, null).GetMixedTypesClient().SendAsync(new MixedTypesCases(ModelReaderWriter.Write(new Cat("test")),
+                BinaryData.FromObjectAsJson("a"),
+                BinaryData.FromObjectAsJson(2),
+                BinaryData.FromObjectAsJson(true)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        private static void AssertEqual(BinaryData source, BinaryData target)
+        {
+            var sourceData = source.ToArray();
+            var targetDate = target.ToArray();
+            CollectionAssert.AreEqual(sourceData, targetDate);
+        }
```

Comment:
could we remove this method and change all its usage to `BinaryDataAssert.AreEqual`? We recently created this new helper

Reply:
fixed